discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -0,0 +1,196 @@
+                let url = Url::parse(&hd.url).map_err(|e| Error::TypeConversion(e.into()))?;
+                Ok(HistoricDocument {
+                    id: Id::from(hd.document),
+                    url,
+                    snippet: hd.snippet,
+                    title: hd.title,
+                })
+            })
+            .collect()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::document::NewsResource;
+
+    use super::*;
+
+    fn create_historic_documents(n: u32) -> Vec<(HistoricDocument, NewsResource)> {
+        (0..n)
+            .map(|i| {
+                let title = format!("title-{}", i);
+                let snippet = format!("snippet-{}", i);
+                let url = Url::parse("http://example.com").unwrap();
+                (
+                    HistoricDocument {
+                        id: Id::new(),
+                        url: url.clone(),
+                        snippet: snippet.clone(),
+                        title: title.clone(),
+                    },
+                    NewsResource {
+                        title,
+                        snippet,
+                        url,
+                        ..Default::default()
+                    },
+                )
+            })
+            .collect()
+    }
+
+    async fn insert_historic_documents(
+        storage: &SqliteStorage,
+        docs: Vec<(HistoricDocument, NewsResource)>,
+    ) {
+        let mut con = storage.pool.acquire().await.unwrap();
+
+        for doc in docs {
+            sqlx::query("INSERT INTO Document (id) VALUES (?)")
+                .bind(doc.0.id.as_uuid())
+                .execute(&mut con)
+                .await
+                .unwrap();
+            sqlx::query("INSERT INTO HistoricDocument (document) VALUES (?)")
+                .bind(doc.0.id.as_uuid())
+                .execute(&mut con)
+                .await
+                .unwrap();
+            sqlx::query("INSERT INTO NewsResource (document, url, snippet, title, topic, datePublished, source) VALUES (?, ?, ?, ?, ?, ?, ?)")
+                .bind(doc.0.id.as_uuid())
+                .bind(doc.1.url.as_str())
+                .bind(doc.1.snippet)
+                .bind(doc.1.title)
+                .bind(doc.1.topic)
+                .bind(doc.1.date_published.to_string())
+                .bind(doc.1.source_domain)
+                .execute(&mut con)
+                .await
+                .unwrap();
+        }
+    }
+
+    #[tokio::test]
+    async fn test_fetch_history() {
+        let storage = SqliteStorage::connect("sqlite::memory:").await.unwrap();
+        storage.init_database().await.unwrap();
+        let history = storage.fetch_history().await.unwrap();
+        assert!(history.is_empty());
+
+        let docs = create_historic_documents(10);
+        insert_historic_documents(&storage, docs.clone()).await;
+
+        let history = storage.fetch_history().await.unwrap();
+        assert_eq!(docs.len(), history.len());
```

Comment:
:art: i'd revert the argument order to match the usual `assert_eq!(actual, expected)`. similarly for the iterated assertions below.
```suggestion
        assert_eq!(history.len(), docs.len());
```